RecoLocalTracker/SiStripClusterizer/plugins/alpaka/SiStripClustersToLegacyAlgo.dev.cc

```diff
@@ -0,0 +1,189 @@
+
+#include <alpaka/alpaka.hpp>
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/workdivision.h"
+
+#include "SiStripClustersToLegacyAlgo.h"
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE::sistrip {
+  using namespace ::sistrip;
+  using namespace sistripConverter;
+  using namespace cms::alpakatools;
+
+  class siStripConvKer_fillClCollSlim {
+  public:
+    template <typename TAcc, typename = std::enable_if_t<alpaka::isAccelerator<TAcc>>>
+    ALPAKA_FN_HOST_ACC void operator()(TAcc const& acc,
+                                       const uint32_t nbClusterCandidates,
+                                       SiStripClusterConstView fullCollection,
+                                       SiStripClustersSlimView slimCollection) const {
+      //
+      // for (uint32_t i=0; i<nbClusterCandidates; ++i) {
+      for (auto i : uniform_elements(acc, nbClusterCandidates)) {
+        const bool write = fullCollection.candidateAccepted(i);
+        // if (i<100 || i> (nbClusterCandidates - 100)) {
+        //   printf(
+        //     "#fccs,%i,%i,%i,%i,%i,%i,%f,%f,%i\n",
+        //     i,
+        //     fullCollection.clusterIndex(i),
+        //     fullCollection.clusterSize(i),
+        //     fullCollection.clusterDetId(i),
+        //     fullCollection.firstStrip(i),
+        //     fullCollection.candidateAccepted(i),
+        //     fullCollection.barycenter(i),
+        //     fullCollection.charge(i),
+        //     fullCollection.candidateAcceptedPrefix(i)
+        //   );
+        // }
+        if (write) {
+          const uint32_t idx = fullCollection.candidateAcceptedPrefix(i) - 1;
+          slimCollection.clusterIndex(idx) = fullCollection.clusterIndex(i);
+          slimCollection.clusterSize(idx) = fullCollection.clusterSize(i);
+          slimCollection.clusterDetId(idx) = fullCollection.clusterDetId(i);
+          slimCollection.firstStrip(idx) = fullCollection.firstStrip(i);
+          slimCollection.barycenter(idx) = fullCollection.barycenter(i);
+          slimCollection.charge(idx) = fullCollection.charge(i);
```

Comment:
it should be slightly more efficient to do
```suggestion
          const auto full = fullCollection[i];
          const uint32_t idx = fullCollection.candidateAcceptedPrefix(i) - 1;
          auto slim = slimCollection[idx];
          slim.clusterIndex() = full.clusterIndex();
          slim.clusterSize() = full.clusterSize();
          slim.clusterDetId() = full.clusterDetId();
          slim.firstStrip() = full.firstStrip();
          slim.barycenter() = full.barycenter();
          slim.charge() = full.charge();
```